// gatsby-plugin-loadable-components-ssr/gatsby-node.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

exports.__esModule = true;
exports.onPostBuild = exports.onCreateBabelConfig = exports.onCreateWebpackConfig = void 0;

var _fs = require("fs");

var _path = _interopRequireDefault(require("path"));

var _webpackPlugin = _interopRequireDefault(require("@loadable/webpack-plugin"));

const loadableStatsName = "loadable-stats-build-javascript.json";

const statsPath = _path.default.join(process.cwd(), `/public/${loadableStatsName}`);

const isWin = process.platform === "win32";

const onCreateWebpackConfig = ({
  actions,
  stage
}) => {
  if (stage === "build-javascript" || stage === "develop" || stage === "develop-html") {
    actions.setWebpackConfig({
      plugins: [new _webpackPlugin.default({
        filename: isWin ? loadableStatsName : statsPath,
        writeToDisk: true
      })]
    });
  }
};

exports.onCreateWebpackConfig = onCreateWebpackConfig;

const onCreateBabelConfig = ({
  actions
}) => {
  actions.setBabelPlugin({
    name: "@loadable/babel-plugin"
  });
};

exports.onCreateBabelConfig = onCreateBabelConfig;

const onPostBuild = () => {
  try {
    // Clean after ourselves
    (0, _fs.unlinkSync)(statsPath);
  } catch (e) {// Ignore as we just want to avoid crashing a build.
  }
};

exports.onPostBuild = onPostBuild;
